packages/lib/services/e2ee/EncryptionService.ts

```diff
@@ -394,6 +416,26 @@ export default class EncryptionService {
 				}
 			},
 
+			// New encryption method powered by native crypto libraries(node:crypto/react-native-quick-crypto). Using AES-256-GCM and pbkdf2
+			// The master key is not directly used. A new data key is generated from the master key and a 256 bits random salt to prevent nonce reuse problem
```

Comment:
**🟣 Commenting**: It may make sense to document this and other choices related to the new encryption methods in a file in `/readme/dev` (see the existing files in [`/readme/dev/spec/e2ee`](https://github.com/laurent22/joplin/tree/dev/readme/dev/spec/e2ee)).

Reply:
Would it be better to make a separate PR for the documents?

Reply:
I think a separate PR would make sense.

There are a few potential benefits to including this in the main pull request:
- **Code review**: May give code reviewers an overview of significant changes and design decisions.
    - This could also be accomplished by linking the documents PR to this one.
- **Commit history**: Associates document changes with relevant code changes in Joplin's commit history. 

However, including the documents in this pull request also has drawbacks:
- More changes to be reviewed in this pull request.
- More work before this pull request is ready to be merged.